pkg/beholder/example_test.go

```diff
@@ -0,0 +1,89 @@
+package beholder_test
+
+import (
+	"context"
+	"log"
+	"math/rand"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/proto"
+
+	// chainlink-common
+	beholder "github.com/smartcontractkit/chainlink-common/pkg/beholder/global"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder/pb"
+)
+
+func ExampleBeholderCustomMessage() {
+	beholderConfig := beholder.NewConfig()
+
+	// Bootstrap Beholder Client
+	err := beholder.Bootstrap(beholderConfig, errorHandler)
+	if err != nil {
+		log.Fatalf("Error bootstrapping Beholder: %v", err)
+	}
+
+	// Define a custom protobuf payload to emit
+	payload := &pb.TestCustomMessage{
+		BoolVal:   true,
+		IntVal:    42,
+		FloatVal:  3.14,
+		StringVal: "Hello, World!",
+	}
+	payloadBytes, err := proto.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Failed to marshal protobuf")
+	}
+
+	// Emit the custom message anywhere from application logic
+	for range 10 {
+		err := beholder.Emit(context.Background(), payloadBytes,
+			"beholder_data_schema", "/custom-message/versions/1", // required
+			"beholder_data_type", "custom_message",
+			"foo", "bar",
+		)
+		if err != nil {
+			log.Printf("Error emitting message: %v", err)
+		}
+	}
+	// Output:
+}
+
+func ExampleBeholderMetricTraces() {
+	beholderConfig := beholder.NewConfig()
+
+	// Bootstrap Beholder Client
+	err := beholder.Bootstrap(beholderConfig, errorHandler)
+	if err != nil {
+		log.Fatalf("Error bootstrapping Beholder: %v", err)
+	}
+
+	// Define a new counter
+	counter, err := beholder.Meter().Int64Counter("custom_message.count")
+	if err != nil {
+		log.Fatalf("failed to create new counter")
+	}
+
+	// Define a new gauge
+	gauge, err := beholder.Meter().Int64Gauge("custom_message.gauge")
+	if err != nil {
+		log.Fatalf("failed to create new gauge")
+	}
+
+	// Use the counter and gauge for metrics within application logic
+	counter.Add(context.Background(), 1)
+	gauge.Record(context.Background(), rand.Int63n(101))
+
+	// Create a new trace span
+	_, rootSpan := beholder.Tracer().Start(context.Background(), "foo", trace.WithAttributes(
```

Comment:
for parity of system behavior, I think it's worth it to drill context through `ExampleBeholderMetricTraces()` and use that inherited context here.

Reply:
https://github.com/smartcontractkit/chainlink-common/pull/694/commits/d0a857af3419bae08cbc01c1fbb6c6f9d26db407